python_modules/libraries/dagster-components/dagster_components/core/component.py

```diff
@@ -204,12 +207,57 @@ def get_component_types_in_module(
 T = TypeVar("T")
 
 
+@dataclass
+class MultiComponentsLoadContext:
```

Comment:
I think this have name that is more suited to its purpose. 

Is this instantiated once per code location? Or once per module as we recursively descend?

Reply:
e.g. `DefinitionsCache` would be more clear

Reply:
I like `DefinitionsModuleCache`. Updated.